src/transformers/testing_utils.py

```diff
@@ -3258,29 +3272,29 @@ def is_default(key: DeviceProperties) -> bool:
     def score(key: DeviceProperties, other: DeviceProperties) -> int:
         """
         Returns score indicating how similar two instances of the `Properties` tuple are.
-        Points are calculated using bits, but documented as int.
         Rules are as follows:
-            * Matching `type` gives 8 points.
-            * Semi-matching `type`, for example cuda and rocm, gives 4 points.
-            * Matching `major` (compute capability major version) gives 2 points.
-            * Default expectation (if present) gives 1 points.
+            * Matching `type` adds one point, semi-matching `type` adds half a point (e.g. cuda and rocm).
+            * If types match, matching `major` adds another point, and then matching `minor` adds another.
+            * Default expectation (if present) is worth 0.1 point to distinguish it from a straight-up zero.
```

Comment:
Any reason to change the rubric?

Reply:
@ydshieh  for second look

Reply:
Imo the old system's way of couting point made sense for bytes but since everything was converted to int in the end it made things a bit convoluted. New system makes it a bit more clear and does not try to match major or minor version across architecture because they don't have the same convetions.

Reply:
I have to say I didn't dive into the details when @yao-matrix implemented this part.

But in the new system introduce here, 

actual one: ("cuda", 9, 2)

candidate 1: ("cuda", 9, 1)
candidate 2: ("cuda", 10, 2)

candidate 1 and 2 will have the same score? Not 100% sure for this.

But I am good to go for it. 

@yao-matrix Do you see any downside?

Reply:
Ah nice

(I only read the docstring, you got me being lazy 😅 )